addons/sale_loyalty/tests/test_program_with_code_operations.py

```diff
@@ -411,3 +429,109 @@ def test_edit_and_reapply_promotion_program(self):
         self._apply_promo_code(order, 'test')
         # But the above line should not add any reward
         self.assertEqual(len(order.order_line), 2, "You should get a discount line") # product + discount
+
+    def test_apply_max_discount_from_multiple_rewards_1(self):
+        """ Test applying the maximum reward discount from multiple rewards when the applied
+            coupon discount is lower.
+
+        1. Create a two loyalty program of type `coupons`.
+        2. Add multiple rewards to the second program.
+        3. Generate a coupon for each program.
+        2. Create a sale order and add a product to it.
+        3. Apply the Coupon to the sale order.
+        4. Try to apply the second coupon with multiple rewards.
+        5. Reward with best discount will be shown.
+        """
+        self.code_promotion_program_with_discount.rule_ids.unlink()
+        coupon_1 = self._generate_coupons(self.code_promotion_program_with_discount)
+        coupon_2 = self._generate_coupons(self.discount_with_multi_rewards)
+
+        order = self.empty_order
+        order.write({'order_line': [
+            (0, 0, {
+                'product_id': self.product_A.id,
+                'name': '1 Product A',
+                'product_uom': self.uom_unit.id,
+                'product_uom_qty': 1.0,
+            })
+        ]})
+
+        self._apply_promo_code(order, coupon_1.code)
+        self.assertEqual(len(order.order_line.ids), 2)
+        self._apply_promo_code(order, coupon_2.code)
+        self.assertEqual(order.order_line[1].price_unit, -20.0)
+
+        expected_total = order.order_line[0].price_total * 0.80
```

Comment:
Actually I think expected_total should be amount_total * 0,8 before any code was applied. Because this approach only works if we have one product in order.